standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -468,6 +459,25 @@ void findUnknownPartitions(Table table, Set<Path> partPaths, byte[] filterExp,
         }
       }
     }
+    
+    Set<Path> partPathsInMS = partPaths;
+    // don't want the table dir
+    partPathsInMS.remove(tablePath);
+    // remove partition paths in partPathsInMS, to getPartitionsNotOnFs
+    partPathsInMS.removeAll(allPartDirs);
```

Comment:
allPartDirs here contains all partition path which are there in FileSystem but not in metastore. Removing it from partPathsInMS doesn't make sense.

I think this operation should be done before allPartDirs.removeAll(partPaths);